src/luarocks/signing.lua

```diff
@@ -0,0 +1,49 @@
+local signing = {}
+
+local cfg = require("luarocks.core.cfg")
+local fs = require("luarocks.fs")
+
+local function get_gpg()
+   local vars = cfg.variables
+   local gpg = vars.GPG
+   local gpg_ok, err = fs.is_tool_available(gpg, "gpg")
+   if not gpg_ok then
+      return nil, err
+   end
+   return gpg
+end
+
+function signing.signature_url(url)
+   return url .. ".asc"
+end
+
+function signing.sign_file(file)
+   local gpg, err = get_gpg()
+   if not gpg then
+      return nil, err
+   end
+   
+   local sigfile = file .. ".asc"
+   fs.delete(sigfile)
```

Comment:
Would it be better to throw an error if the signature already exists?

Reply:
if we don't delete, gpg asks if we want to override. Since entering the passphrase implies an interaction anyway, I just removed the `fs.delete`.